ProductAPI helpers do not catch all errors properly

```
[2023-09-28 12:44:46 +0000] [11] [ERROR] Exception in ASGI application
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/aiohttp/connector.py", line 980, in _wrap_create_connection
    return await self._loop.create_connection(*args, **kwargs)  # type: ignore[return-value]  # noqa
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "uvloop/loop.pyx", line 2039, in create_connection
  File "uvloop/loop.pyx", line 2016, in uvloop.loop.Loop.create_connection
ConnectionRefusedError: [Errno 111] Connection refused

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/uvicorn/protocols/http/httptools_impl.py", line 419, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/uvicorn/middleware/proxy_headers.py", line 78, in __call__
    return await self.app(scope, receive, send)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/fastapi/applications.py", line 292, in __call__
    await super().__call__(scope, receive, send)
  File "/usr/local/lib/python3.11/site-packages/starlette/applications.py", line 122, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/usr/local/lib/python3.11/site-packages/starlette/middleware/errors.py", line 184, in __call__
    raise exc
  File "/usr/local/lib/python3.11/site-packages/starlette/middleware/errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File "/usr/local/lib/python3.11/site-packages/rasenmaeher_api/db/middleware.py", line 107, in __call__
    await self.app(scope, receive, send)
  File "/usr/local/lib/python3.11/site-packages/starlette/middleware/exceptions.py", line 79, in __call__
    raise exc
  File "/usr/local/lib/python3.11/site-packages/starlette/middleware/exceptions.py", line 68, in __call__
    await self.app(scope, receive, sender)
  File "/usr/local/lib/python3.11/site-packages/fastapi/middleware/asyncexitstack.py", line 20, in __call__
    raise e
  File "/usr/local/lib/python3.11/site-packages/fastapi/middleware/asyncexitstack.py", line 17, in __call__
... (1 KB left)
````

The backend call error should be logged but not raised.

--- a/.bumpversion.cfg
+++ b/.bumpversion.cfg
@@ -1,5 +1,5 @@
 [bumpversion]
-current_version = 0.6.1
+current_version = 0.6.2
 commit = False
 tag = False
 

--- a/Dockerfile
+++ b/Dockerfile
@@ -187,5 +187,8 @@ RUN apt-get update && apt-get install -y zsh jq \
     && sh -c "$(curl -fsSL https://raw.githubusercontent.com/ohmyzsh/ohmyzsh/master/tools/install.sh)" \
     && echo "source /root/.profile" >>/root/.zshrc \
     && pip3 install git-up \
+    # Map the special names to docker host internal ip because 127.0.0.1 is *container* localhost on login
+    && echo "sed 's/.*localmaeher.*//g' /etc/hosts >/etc/hosts.new && cat /etc/hosts.new >/etc/hosts" >>/root/.profile \
+    && echo "echo \"\$(getent hosts host.docker.internal | awk '{ print $1 }') fake.localmaeher.pvarki.fi\" >>/etc/hosts" >>/root/.profile \
     && true
 ENTRYPOINT ["/bin/zsh", "-l"]

--- a/docker/entrypoint.sh
+++ b/docker/entrypoint.sh
@@ -1,5 +1,8 @@
 #!/bin/bash -l
 set -e
+# Make sure fakeproduct api endpoint points to correct IP, 127.0.01 is this containers localhost...
+sed 's/.*localmaeher.*//g' /etc/hosts >/etc/hosts.new && cat /etc/hosts.new >/etc/hosts
+echo "$(getent hosts host.docker.internal | awk '{ print $1 }') fake.localmaeher.pvarki.fi" >>/etc/hosts
 if [ "$#" -eq 0 ]; then
   # FIXME: can we know the traefik/nginx internal docker ip easily ?
   exec gunicorn "rasenmaeher_api.web.application:get_app()" --bind 0.0.0.0:8000 --forwarded-allow-ips='*' -w 4 -k uvicorn.workers.UvicornWorker

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "rasenmaeher_api"
-version = "0.6.1"
+version = "0.6.2"
 description = "python-rasenmaeher-api"
 authors = [
     "Aciid <703382+Aciid@users.noreply.github.com>",

--- a/src/rasenmaeher_api/__init__.py
+++ b/src/rasenmaeher_api/__init__.py
@@ -1,2 +1,2 @@
 """ python-rasenmaeher-api """
-__version__ = "0.6.1"  # NOTE Use `bump2version --config-file patch` to bump versions correctly
+__version__ = "0.6.2"  # NOTE Use `bump2version --config-file patch` to bump versions correctly

--- a/src/rasenmaeher_api/console.py
+++ b/src/rasenmaeher_api/console.py
@@ -18,6 +18,7 @@ from rasenmaeher_api.db import base as dbbase
 from rasenmaeher_api.db.config import DBConfig
 from rasenmaeher_api.db.middleware import DBWrapper
 from rasenmaeher_api.web.application import get_app_no_init
+from rasenmaeher_api.db.base import init_db, bind_config
 
 
 LOGGER = logging.getLogger(__name__)
@@ -132,9 +133,16 @@ def add_test_users(ctx: click.Context) -> None:
     """
     Create the test users defined in testhelpers.create_test_users
     """
-    ret = create_test_users()
-    click.echo(pprint.pformat(ret))
-    ctx.exit(0)
+
+    async def call_testusers() -> int:
+        """Start db connection, call the helper"""
+        await bind_config()
+        await init_db()
+        ret = await create_test_users()
+        click.echo(pprint.pformat(ret))
+        return 0
+
+    ctx.exit(ctx.obj["loop"].run_until_complete(call_testusers()))
 
 
 def rasenmaeher_api_cli() -> None:

--- a/src/rasenmaeher_api/prodcutapihelpers.py
+++ b/src/rasenmaeher_api/prodcutapihelpers.py
@@ -57,11 +57,10 @@ async def _method_to_all_products(
     LOGGER.debug("data={}".format(data))
     async with session as client:
         for name, conf in manifest["products"].items():
-            url = f"{conf['api']}{url_suffix}"
-            LOGGER.debug("calling {}({})".format(methodname, url))
-
-            resp = await getattr(client, methodname)(url, json=data)
             try:
+                url = f"{conf['api']}{url_suffix}"
+                LOGGER.debug("calling {}({})".format(methodname, url))
+                resp = await getattr(client, methodname)(url, json=data)
                 resp.raise_for_status()
                 payload = await resp.json()
                 LOGGER.debug("payload={}".format(payload))

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -86,7 +86,12 @@ def session_env_config(  # pylint: disable=R0915
                         "api": f"https://fake.localmaeher.pvarki.fi:{fakeproduct_port}/",
                         "uri": "https://fake.localmaeher.pvarki.fi:844/",  # Not actually there
                         "certcn": "fake.localmaeher.pvarki.fi",
-                    }
+                    },
+                    "nonexistent": {
+                        "api": "https://localhost:4657/",
+                        "uri": "https://nonexistent.localmaeher.pvarki.fi:844/",  # Not actually there
+                        "certcn": "nonexistent.localmaeher.pvarki.fi",
+                    },
                 },
             }
         )

--- a/tests/test_rasenmaeher_api.py
+++ b/tests/test_rasenmaeher_api.py
@@ -10,7 +10,7 @@ from rasenmaeher_api import __version__
 
 def test_version() -> None:
     """Make sure version matches expected"""
-    assert __version__ == "0.6.1"
+    assert __version__ == "0.6.2"
 
 
 @pytest.mark.asyncio

